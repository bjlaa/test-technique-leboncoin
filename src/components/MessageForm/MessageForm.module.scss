@import '../../variables';

.messageForm {
	text-align: left;
	margin-top: 10px;
	&__text {
		padding: 10px 0px;
		&__textarea {
			width: 100%;
			min-height: 70px;
			margin: 0 auto;
			border: solid 1px lightgrey;
			border-radius: 3px;
			display: block;
			font-size: 10pt;
		}
	}
	&__private {
		padding: 10px 0px;
		text-align: right;
		display: flex;
		align-items: flex-end;
		justify-content: flex-end;
		font-size: 10pt;
		&__label {
			margin-right: 10px;
		}
		&__input {

		}
	}
	&__submitButton {
    background: #282c34;
    border: none;
    border-radius: 3px;
    padding: 5px 10px;
    font-size: 10pt;
    color: white;
    display: block;
    margin-left: auto;
    margin-top: 10px;
	}
}
@media (min-width: $min-width-tablet) {
  .messageForm {
    width: 50%;
    align-self: flex-end;
		&__text {
			align-items: flex-end;
			&__textarea {
				max-width: 80%;
				margin-right: 0;
			}
		}
		&__submitButton {

		}
  }
}