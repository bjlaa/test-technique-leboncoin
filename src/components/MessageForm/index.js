import React from 'react';

import styles from './MessageForm.module.scss';

class MessageForm extends React.Component {
	constructor(props) {
		super(props);

		this.textarea = React.createRef();
		this.privateCheckbox = React.createRef();
	}

	// Handles the submit of our form
	handleSubmit = (event) => {
		event.preventDefault();

		const newMessage = {
			text: this.textarea.current.value,
			private: this.privateCheckbox.current.checked
		};

		this.props.createMessage(newMessage);
	};

	render() {
		return (
			<form
				className={styles.messageForm}
				onSubmit={this.handleSubmit}
			>
				<div className={styles.messageForm__text}>
					<label htmlFor="messageForm__text__textarea">
						Type in your message:
					</label>
					<textarea
						className={styles.messageForm__text__textarea}
						name=""
						id="messageForm__text__textarea"
						cols="30"
						rows="10"
						ref={this.textarea}
					/>
				</div>
				<div className={styles.messageForm__private}>
					<label htmlFor="messageForm__private__input">
						Private message:
					</label>
					<input
						className={styles.messageForm__private__input}
						type="checkbox"
						id="messageForm__private__input"
						ref={this.privateCheckbox}
					/>
				</div>
				<button
					type="submit"
					className={styles.messageForm__submitButton}
				>
					Send
				</button>
			</form>
		);
	}
}

export default MessageForm;
