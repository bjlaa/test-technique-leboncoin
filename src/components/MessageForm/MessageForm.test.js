import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MessageForm from './index';

configure({ adapter: new Adapter() });

describe('MessageForm', () => {
	it('should match its snapshot', () => {
		const component = renderer.create(<MessageForm />);

		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should create a new message on form submit', () => {
		const spy = jest.fn();

		// We pass a spy function as the createMessage prop
		const component = mount(<MessageForm createMessage={spy} />);

		component.find('form').simulate('submit');

		expect(spy).toHaveBeenCalled();
	});

	it('should empty the form textarea and clear the private checkbox upon form submit', () => {
		const component = mount(<MessageForm createMessage={() => {}}/>);

		// We test that the textarea empties itself by setting it value to a string
		component.find('textarea').text('test');
		// Similarly we set the checkbox to true to check if it is reinitialized
		component.find('input').simulate('change', { target: { checked: true }});

		component.find('form').simulate('submit');

		expect(component.find('textarea').text()).toEqual('');
		expect(component.find('input').instance().checked).toEqual(false);
	});
});
