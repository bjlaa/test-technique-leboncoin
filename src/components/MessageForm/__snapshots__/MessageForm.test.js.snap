// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageForm should match its snapshot 1`] = `
<form
  className="messageForm"
  onSubmit={[Function]}
>
  <div
    className="messageForm__text"
  >
    <label
      htmlFor="messageForm__text__textarea"
    >
      Type in your message:
    </label>
    <textarea
      className="messageForm__text__textarea"
      cols="30"
      id="messageForm__text__textarea"
      name=""
      rows="10"
    />
  </div>
  <div
    className="messageForm__private"
  >
    <label
      htmlFor="messageForm__private__input"
    >
      Private message:
    </label>
    <input
      className="messageForm__private__input"
      id="messageForm__private__input"
      type="checkbox"
    />
  </div>
  <button
    className="messageForm__submitButton"
    type="submit"
  >
    Send
  </button>
</form>
`;
